doc/source/groups_of_atoms.rst

```diff
@@ -64,6 +64,66 @@ Similarly, an :class:`~MDAnalysis.core.groups.Atom` has direct knowledge of the
+we present two use cases showing commonly used applications.
+
+---------------------------------------
+Use case: Sequence of residues by chain
+---------------------------------------
+Let's assume that we have a PDB file, which contains both ATOM and HETATM record types, for protein and small molecule,
+respectively. In order to select only ATOM record types and get a list of residues by chain, one needs to call:
+
+.. code-block:: python
+
+    u = mda.Universe(pdb, format="PDB")
+    for seg in u.segments:
+        p_seg = seg.atoms.select_atoms("record_type ATOM")
+        r_list = p_seg.residues.resnames
+        print(r_list)
+
+This can be used later on to check whether the residue names belong, for example, to the list of standard residues.
+However, it would be **incorrect** to use:
+
+.. code-block:: python
+
+    selected_atoms = u.select_atoms("record_type ATOM")  # CORRECT SYNTAX, BUT INCORRECT RETURN VALUE!
+    residues_wrong = selected_atoms.segments.resnames  # CORRECT SYNTAX, BUT INCORRECT RETURN VALUE!
+    print(residues_wrong)  # CORRECT SYNTAX, BUT INCORRECT RETURN VALUE!
+
+because, although it nicely returns list of residues in every chain (list of lists, so no need to additionally
+loop through segments), it would return all residues, both for ATOM and HETATM record types, because this construct
+means "give me all residue names from segments in which there is at least one of the selected atoms".
+
+----------------------------------------
+Use case: Atoms list grouped by residues
+----------------------------------------
+In another use case, let's assume that we would like to check if all the heavy protein backbone and sidechain atoms
+are present in every residue. We can easily prepare the expected atom names list in every standard residue, but
+we need to obtain the list of atoms in every residue to compare with (using, for example, a symmetrical difference).
+In order to do that, we want to avoid HETATMs, OXT and hydrogen (H*) atoms, and we need to loop over segments (chains)
+and residues:
+
+.. code-block:: python
+
+    u = mda.Universe(pdb, format="PDB")
+    for seg in u.segments:
+        p_seg = seg.atoms.select_atoms("record_type ATOM and not (name H* or name OXT)")
+        for p_res in p_seg.residues:
+            atoms_in_residue = p_seg.select_atoms(f"resid {p_res.resid} and resname {p_res.resname}").names.tolist()
+            print(atoms_in_residue)
+
+However, it would be **incorrect** to use in the body of the first loop:
+
+.. code-block:: python
+
+    atoms_in_residue_wrong = p_seg.residues.names  # CORRECT SYNTAX, BUT INCORRECT RETURN VALUE!
+    print(atoms_in_residue_wrong)  # CORRECT SYNTAX, BUT INCORRECT RETURN VALUE!
+
+because, although it nicely returns list of atoms in every residue, it would return all atoms (including hydrogen ones),
+because this construct means "give me all atom names from residues in which there is at least one of the selected
+atoms".
```

Comment:
This scenario is a bit more involved and I think would be better as a separate, small tutorial, for people who do want to do this task. You could explain why you do not want the "H*" and "OXT" atoms, show where you can get a list of atoms in standard residues, and show the comparison via symmetric difference. You could then link to that tutorial here.

Otherwise this use case assumes a fair bit biological/MD background knowledge (e.g. why avoid HETATM? what is a HETATM vs ATOM?)

Side note, I think you could also do (untested):

```python
segs = u.select_atoms("record_type ATOM and not name H* OXT").split("residue")
print([ag.names for ag in segs])
```

If I recall correctly, residues are always in one segment or another, they cannot be split over segments.

But it is worthwhile going over loops etc. for new users.

Reply:
If it's a self-contained example then it could become a notebook in the [Examples](https://userguide.mdanalysis.org/1.0.0/examples/README.html) section, couldn't it?

Reply:
Excellent idea @orbeckst ! Where is this Examples page you think about? Do you mean adding it here: https://www.mdanalysis.org/MDAnalysisTutorial/atomgroups.html ?

I can add those to Examples and reference from AtomGroup page. Is that OK to add a few lines of PDB file to the example rather than adding it to `MDAnalysis.tests.datafiles` ?

Reply:
No, the MDAnalysisTutorial is outdated and really deprecated.

I mean the Examples section in the User Guide https://userguide.mdanalysis.org/examples/README.html – add a new page there.